phases/codesign.go

```diff
@@ -2,133 +2,175 @@ package phases
 
 import (
 	"fmt"
-	"io"
-	"io/ioutil"
-	"net/http"
-	"os"
 	"path/filepath"
-	"strings"
 
-	"github.com/bitrise-io/go-utils/command"
+	bitriseModels "github.com/bitrise-io/bitrise/models"
+	"github.com/bitrise-io/codesigndoc/codesigndoc"
+	"github.com/bitrise-io/codesigndoc/models"
+	"github.com/bitrise-io/codesigndoc/xcode"
+	envmanModels "github.com/bitrise-io/envman/models"
 	"github.com/bitrise-io/go-utils/log"
+	"github.com/bitrise-io/go-utils/sliceutil"
+	"github.com/bitrise-io/goinp/goinp"
 )
 
+// CodesignResultsIOS ...
+type CodesignResultsIOS struct {
+	certificates         models.Certificates
+	provisioningProfiles []models.ProvisioningProfile
+}
+
+// CodesignResultAndroid ...
+type CodesignResultAndroid struct {
+	KeystorePath, Password, Alias, KeyPassword string
+}
+
 // CodesignResult ...
 type CodesignResult struct {
-	KeystorePath, Password, Alias, KeyPassword string
-	ProfilePaths, CertificatePaths             []string
+	Android CodesignResultAndroid
+	IOS     CodesignResultsIOS
 }
 
-const (
-	platformIOS        = "iOS"
-	platformAndroid    = "Android"
-	platformBoth       = "iOS and Android"
-	codesignExportsDir = "codesigndoc_exports"
-)
+// Project types: "web", "macos", "other", "": is not suppored
 
-func getPlatform(projectType string) string {
-	switch projectType {
-	case "ios":
-		return platformIOS
-	case "android":
-		return platformAndroid
-	case "xamarin", "flutter", "cordova", "ionic", "react-native":
-		return platformBoth
-	case "web", "macos":
-		// not supported
-	}
-	// other or if no project type specified
-	return ""
+var codesignBothPlatforms = []string{"xamarin", "flutter", "cordova", "ionic", "react-native"}
+
+func isIOSCodesign(projectType string) bool {
+	return projectType == "ios" || sliceutil.IsStringInSlice(projectType, codesignBothPlatforms)
+}
+
+func isAndroidCodesign(projectType string) bool {
+	return projectType == "android" || sliceutil.IsStringInSlice(projectType, codesignBothPlatforms)
 }
 
 // AutoCodesign ...
-func AutoCodesign(projectType string) (CodesignResult, error) {
-	log.Donef("Project type: %s", projectType)
+func AutoCodesign(bitriseYML bitriseModels.BitriseDataModel, searchDir string) (CodesignResult, error) {
+	log.Donef("Project type: %s", bitriseYML.ProjectType)
 	fmt.Println()
 
+	if !isIOSCodesign(bitriseYML.ProjectType) && !isAndroidCodesign(bitriseYML.ProjectType) {
+		log.Infof("Unsupported project type for automatic codesigning export.")
```

Comment:
could be a warning and would be great to include the actual and supported project types